docs/content/development/extensions-core/bloom-filter.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: doc_page
+---
+
+# Druid Bloom Filter
+
+Make sure to [include](../../operations/including-extensions.html) `druid-bloom-filter` as an extension.
+
+BloomFilter is a probabilistic data structure for set membership check. 
+Following are some characterstics of BloomFilter 
+- BloomFilters are highly space efficient when compared to using a HashSet.
+- Because of the probabilistic nature of bloom filter false positive (element not present in bloom filter but test() says true) are possible
+- false negatives are not possible (if element is present then test() will never say false). 
+- The false positive probability is configurable (default: 5%) depending on which storage requirement may increase or decrease. 
+- Lower the false positive probability greater is the space requirement.
+- Bloom filters are sensitive to number of elements that will be inserted in the bloom filter.
+- During the creation of bloom filter expected number of entries must be specified.If the number of insertions exceed the specified initial number of entries then false positive probability will increase accordingly.
+
+Internally, this implementation of bloom filter uses Murmur3 fast non-cryptographic hash algorithm.
+
+### Json Representation of Bloom Filter
+```json
+{
+  "type" : "bloom",
+  "dimension" : <dimension_name>,
+  "bloomKFilter" : <serialized_bytes_for_BloomKFilter>,
+  "extractionFn" : <extraction_fn>
+}
+```
+
+|Property                 |Description                   |required?                           |
+|-------------------------|------------------------------|----------------------------------|
+|`type`                   |Filter Type. Should always be `bloom`|yes|
+|`dimension`              |The dimension to filter over. | yes |
+|`bloomKFilter`           |Binary representation of `org.apache.hive.common.util.BloomKFilter`| yes |
```

Comment:
nits: 
"Base64 encoded Binary representation....." ?

Also, It appears to me that this module(and property) are really usable only if someone was using hive util `BloomKFilter` class on the client side because serialized representation is specific to that class and not something standard, so maybe we could add "hive" in the names everywhere e.g. `hive-bloom-filter` module and this property being `hiveUtilBloomKFilter` .
or I'm missing something :)

Reply:
updated, yes, its true that the implementation is currently shared from hive-storage-api jar, but the implementation as such does not depend on any of the hive specific functionality. 
I initially made this a hive-bloom-filter but later on renamed to bloom-filter based on review comments.

Reply:
I do think it makes sense to not have `hive` in the names. I can imagine evolving this extension to support some non-hive format too, and make it more divorced from hive. Especially since with the aggregator @clintropolis mentioned, people could create serialized bloom filters without really needing to worry about the fact that they came from hive.

Reply:
To be clear, if we did evolve it that way, it should still support the hive format (to work with the original use case from @nishantmonu51). I just think there is a lot of potential for this extension to be useful beyond hive integration.

Reply:
ah, I understand the naming now. ( I knew I was missing something ).